cli-toolchain/cognito-triggers.md

```diff
@@ -0,0 +1,508 @@
+      customConfirmSignIn: false,
+      signedIn: false,
+      user: {},
+    }
+  },
+  mounted: async function () {
+    await this.$Amplify.Auth.currentAuthenticatedUser()
+      .then((user) => {
+        this.user = user;
+        this.signedIn = true;
+      })
+      .catch((e) => {
+        console.log('currentUser error', e)
+      })
+    AmplifyEventBus.$on('authState', info => {
+      this.customConfirmSignIn = info === 'customConfirmSignIn';
+      this.signedIn = info === 'signedIn';
+    });
+    AmplifyEventBus.$on('localUser', user => {
+      this.user = user;
+    });
+  },
+  methods: {
+    onVerify: function (data) {
+      this.$Amplify.Auth.sendCustomChallengeAnswer(this.user, data)
+        .then( (user) => { 
+          AmplifyEventBus.$emit('authState', 'signedIn')
+          return AmplifyEventBus.$emit('localUser', 'user')
+        })
+        .catch(function (err) { console.log('challenge error: ', err) });
+    },           
+  }
+}
+</script>
+```
+
+main.js
+```js
+import Vue from 'vue'
+import App from './App.vue'
+import Amplify, * as AmplifyModules from 'aws-amplify'
+import { AmplifyPlugin } from 'aws-amplify-vue'
+import awsconfig from './aws-exports'
+
+Amplify.configure({
+  Auth: {
+    identityPoolId: awsconfig.aws_cognito_identity_pool_id,
+    region: awsconfig.aws_cognito_region,
+    userPoolId: awsconfig.aws_user_pools_id,
+    userPoolWebClientId: awsconfig.aws_user_pools_web_client_id,
+    authenticationFlowType: 'CUSTOM_AUTH'
+  }
+});
+Vue.use(AmplifyPlugin, AmplifyModules)
+
+new Vue({
+  render: h => h(App)
+}).$mount('#app')
+```
+
+Finally, in public/index.html add the following script:
+```html
+<script src="https://www.google.com/recaptcha/api.js?onload=vueRecaptchaApiLoaded&render=explicit" async defer></script>
+```
+
+### Basic Scaffolding for a Custom Auth Challenge
+
+This template will configure three triggers: [CreateAuthChallenge](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-create-auth-challenge.html), [DefineAuthChallenge](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-define-auth-challenge.html) and [VerifyAuthChallengeResponse](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-verify-auth-challenge-response.html).
+
+It will not, however, provide a fully-formed custom authentication flow. Instead, it will create a 'hello world' custom auth flow skeleton that you can manually edit. The intent of this template is to give you a starting place for building out your own custom auth flow.
+
+### Email Verification Link with Redirect
+
+Cognito allows you to configure your User Pool to send an email to your users when they attempt to register an account. You can configure this email to contain a link to Cognito's Hosted UI where the user's account will be marked as confirmed. However, there is currently no means to redirect the user back to your app from this Hosted UI page.
```

Comment:
Do we need this line? "However, there is currently no means to redirect the user back to your app from this Hosted UI page."

I understand that we're working around this but we shouldn't call out limitations of services in docs.

Reply:
Fair point.  I'm removing this.